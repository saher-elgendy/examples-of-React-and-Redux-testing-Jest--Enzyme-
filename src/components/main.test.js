import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import Main from './main';
import './../setupTests';

describe('Main component', () => {
    let wrapper;
    let mockProps;
    let findByText;
    beforeEach(() => {
        mockProps = {
            data: {
                error: null,
                books: [],
                loading: false
            }
        }

        wrapper = shallow(<Main {...mockProps} />)

        findByText = (wrapper, type, text) => {
            return wrapper.findWhere(node => node.type() === type && node.text() === text)
        }

    });

    it('render Main component correctly', () => {
        expect(shallowToJson(wrapper)).toMatchSnapshot();
    })

    it('show loading message if loading', () => {
        mockProps = {
            data: {
                ...mockProps.data,
                error: null,
                loading: true
            }
        }

        wrapper = shallow(<Main {...mockProps} />)

        const loadingMessage = findByText(wrapper, 'span', '...loading books')
        expect(loadingMessage.exists()).toBeTruthy()
    })

    it('show error message if something went wrong', () => {
        mockProps = {
            data: {
                ...mockProps.data,
                error: 'something went wrong'
            }
        }

        wrapper = shallow(<Main {...mockProps} />)
        const errorMessage = findByText(wrapper, 'span', 'something went wrong');
        expect(errorMessage.exists()).toBeTruthy();
    })
});