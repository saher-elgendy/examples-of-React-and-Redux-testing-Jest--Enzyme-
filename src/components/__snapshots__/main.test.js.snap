// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main component render Main component correctly 1`] = `
<div
  className="App"
>
  <header
    data-testid="header"
  >
    <h2>
      React Testing
    </h2>
    <Connect(Bag) />
  </header>
  <BookStore
    books={Array []}
  />
</div>
`;
